src/bson-encode.c

```diff
@@ -351,7 +351,7 @@ static void php_phongo_bson_append(bson_t *bson, php_phongo_bson_flags_t flags,
 			if (bson_utf8_validate(Z_STRVAL_P(entry), Z_STRLEN_P(entry), true)) {
 				bson_append_utf8(bson, key, key_len, Z_STRVAL_P(entry), Z_STRLEN_P(entry));
 			} else {
-				phongo_throw_exception(PHONGO_ERROR_UNEXPECTED_VALUE TSRMLS_CC, "Got invalid UTF-8 value serializing '%s'", Z_STRVAL_P(entry));
+				phongo_throw_exception(PHONGO_ERROR_UNEXPECTED_VALUE TSRMLS_CC, "Detected invalid UTF-8 in \"%s\" for fieldname \"%s\"", Z_STRVAL_P(entry), key);
```

Comment:
I would probably suggest to put the field name first in the error message, as if the value includes for example a null char, it might get cut off.

Reply:
The current order was taken from other error messages, but we can easily swap it for this exception.

Reply:
> value includes for example a null char

PHP does allow null characters in exception messages:

```
>>> throw new Exception("foo\0bar")
Exception with message 'foobar'
```

Are you more concerned with logging the messages outside of PHP?